components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
-		errorMSG.append("\"variables\":").append(gson.toJson(msisdnList)).append("}}");
+		dto.setAppId(app);
+		dto.setApiID(api);
+		dto.addMsisdn(msisdn);
+		dto.setAction(action);
+		dto.setServiceProvider(sp);
 
-		gson = new Gson();
-
-		if (userID != null && msisdnList != null) {
-			msisdnList = removeNullMsisdnValues(msisdnList);
-
-			try {
-
-				WhiteListDTO whiteListDTO = new WhiteListDTO();
-				whiteListDTO.setApiID(apiId);
-				whiteListDTO.setApplicationID(appId);
-				whiteListDTO.setUserMSISDN(msisdnList);
-				whiteListDTO.setValidationRegex(whiteListReq.getValidationRegex());
-				whiteListDTO.setValidationPrefixGroup(whiteListReq.getValidationPrefixGroup());
-				whiteListDTO.setValidationDigitsGroup(whiteListReq.getValidationDigitsGroup());
-				//whiteListDTO.setSubscriptionID(userID);
-
-				blackListWhiteListService.whiteListSubscription(whiteListDTO);
-
-				StringBuilder succMSG = new StringBuilder();
+		try {
+			blackListWhiteListService.remove(dto);
 
-				succMSG.append("{\"Success\":{\"messageId\":\"Whitelist Numbers\",".intern());
-				succMSG.append("\"text\":\" Whitelist Numbers Successfully Added to the system \",".intern());
-				succMSG.append("\"variables\":").append(gson.toJson(msisdnList)).append("}}");
+			return Response.status(Response.Status.NO_CONTENT).build();
+		} catch (BusinessException e) {
+			LOG.error("Error removing number", e);
 
-				return Response.status(Response.Status.OK).entity(succMSG.toString()).build();
-			}catch( InvalidMSISDNException ex){
-				StringBuilder errorMessage=new StringBuilder("{ \"Failed\": { \"messageId\": \"Whitelist Numbers\", \"text\": \"Whitelist Numbers could not be added to the system\", \"variables\":  \""+ex.getErrorType().toString()+"\" } }");
-				return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage.toString()).build();
-			}
-			catch (BusinessException ex) {
-				StringBuilder errorMessage=new StringBuilder("{ \"Failed\": { \"messageId\": \"Whitelist Numbers\", \"text\": \"Whitelist Numbers could not be added to the system\", \"variables\": \""+ex.getErrorType().toString()+"\" } }");
-				return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage.toString()).build();
-			} catch (Exception ex) {
-				StringBuilder errorMessage=new StringBuilder("{ \"Failed\": { \"messageId\": \"Whitelist Numbers\", \"text\": \"Whitelist Numbers could not be added to the system\", \"variables\": \""+ex.getMessage()+"\" } }");
-				return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage.toString()).build();
-			}
-		} else {
-			return Response.status(Response.Status.BAD_REQUEST).entity(errorMSG.toString()).build();
+			return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getErrorType()).build();
 		}
 	}
 
-
-
 	@GET
 	@Path("/subscribers")
-	@Produces("application/json")
+	@Produces(MediaType.APPLICATION_JSON)
 	public Response getAllSubscribers() throws SQLException, BusinessException {
```

Comment:
remove the SQLException which is not thrown from the method

Reply:
updated